docs/remote-attestation.md

```diff
@@ -54,12 +54,12 @@ Note over C,P: Exchange Encrypted Data
 
 loop for each invocation
    C-->>C: Generate symmetric Response Key<br>RK
-   C-->>C: Encrypt Response Key and request body with Enclave Public Key<br>Authenticate additional data (plaintext)
+   C-->>C: Encrypt Response Key with Enclave Public Key<br>Encrypt request body with Response Key<br>Authenticate additional data (plaintext)
```

Comment:
Tink [docs](https://developers.google.com/tink/hybrid) say:
```
To encrypt a message, a fresh symmetric key is generated and used to encrypt the plaintext data. The recipient’s public key is used to encrypt the symmetric key only. The final ciphertext consists of the symmetric ciphertext and the encrypted symmetric key.
```

Reply:
Exactly, so we never see the symmetric key directly.

Reply:
But it's still being sent to the server in an encrypted way

Reply:
Yes, I agree. In normal hybrid encryption the sender generates a symmetric secret key and encrypts that with the public key. This is what is described at the top of the Tink hybrid encryption page.

We are using ECIES, which uses Elliptic Curve Diffie-Hellman and HKDF to derive a secret symmetric key. The response key is independent and used for encrypting the response back.

The more technically correct specification will be something like: Generate and ephemeral key pair and encrypt the response key and and request body using the symmetric key derived from the Enclave Public Key and the Ephemeral private key.

The resverse when decrypting derives the same symmetric key from the ephemeral public key (from the sender) and the enclave private key.

Reply:
I think in this diagram we should treat hybrid encryption as a single unit, that produces an opaque ciphertext. The fact that the ciphertext contains a symmetric key is an impl detail of Tink's hybrid encryption scheme.

Reply:
ah, I initially thought that the diagram described the hybrid encryption implementation

Do I understand correctly that on top of the hybrid encryption we introduce a new `Response Key` which is only used to make each response be encrypted with a unique key?

Reply:
> Do I understand correctly that on top of the hybrid encryption we introduce a new Response Key which is only used to make each response be encrypted with a unique key?

Yes that is correct. Ideally that's the same type of key that tink uses within hybrid encryption, but it does not have to be.

Reply:
What kinds of attacks on hybrid encryption do we prevent with this scheme?

Reply:
It's not a mitigation for any attack, it's just that we need a way for the server to encrypt the response back to the client. We could use hybrid encryption again on the way back, but it's overkill, and generating a symmetric key pair is much more expensive than generating a symmetric key.

Reply:
I don't think there is a way of extracting the symmetric key, but maybe I'm wrong. If that's possible via Tink, then I agree it may make sense to reuse it. Though we need to convince ourselves there is no risk of reflection attack (an attacker sending back the request to the client, making it look like a response, even though the attacker cannot decrypt it).

Do you know if it's possible to extract the symmetric key from the payload via Tink functions? (i.e. without manually parsing and decrypting the ciphertext again)

@sidtelang do you have any additional comments?

Reply:
> Do you know if it's possible to extract the symmetric key from the payload via Tink functions? (i.e. without manually parsing and decrypting the ciphertext again)

I'll look into it, but I agree, Tink will probably hide this key inside the implementation

Reply:
It is usually not a good idea to use the same derived key to send the response, as you need to be careful about IV reuse etc.

We could derive a second symmetric key (using different parameters to the HKDF) and just use that to encrypt the response rather than generate a new random key. The HKDF is more computationally expensive and this approach is probably not possible using the Tink abstractions, but it will save a bit of bandwith. For the added complexity I don't think it is worth saving a few bytes.

Reply:
No additional comments in particular: reasonably sure that the public Tink APIs do not let you access the hybrid encryption ephemeral symmetric key and even if it is possible using visibility restricted APIs not sure it's worth pursuing: 
the goal should be a protocol that is easy to reason about, not being able to devise an attack is not sufficient